LibreNMS/Alert/Xmatters.php

```diff
@@ -0,0 +1,121 @@
+<?php
+/**
+ * Xmatters API Transport
+ *
+ * @author Sigurdur Stefan <https://github.com/LoveSkylark>
+ * @copyright 2023 Sigurdur Stefan
+ * @license GPL
+ */
+
+namespace LibreNMS\Alert\Transport;
+
+use LibreNMS\Alert\Transport;
+use LibreNMS\Exceptions\AlertTransportDeliveryException;
+use LibreNMS\Util\Http;
+
+class Xmatters extends Transport
+{
+    public function deliverAlert(array $alert_data): bool
+    {
+        $url = $this->config['xmatters_url'] . '/api/integration/1/functions/' . $this->config['xmatters_function'] . '/triggers';
+
+        $device_url = \Config::get('base_url');
+        $device_groups = \DeviceCache::get($alert_data['device_id'])->groups->pluck('name');
+
+        // Build request data
+        $data = [
+            'properties' => $alert_data,
+            'recipients' => $alert_data['contacts'],
```

Comment:
The contacts field is suspect imo.  It will likely be removed in the future as nothing else is using it.

Reply:
Under stood